src/Aspire.Hosting.Azure.ServiceBus/AzureServiceBusExtensions.cs

```diff
@@ -124,19 +151,381 @@ public static IResourceBuilder<AzureServiceBusResource> AddQueue(this IResourceB
+        => builder.WithBindMount(path ?? $".servicebus/{builder.Resource.Name}", "/data", isReadOnly: false);
+
+    /// <summary>
+    /// Adds a named volume for the data folder to an Azure Service Bus emulator resource.
+    /// </summary>
+    /// <param name="builder">The builder for the <see cref="AzureServiceBusEmulatorResource"/>.</param>
+    /// <param name="name">The name of the volume. Defaults to an auto-generated name based on the application and resource names.</param>
+    /// <returns>A reference to the <see cref="IResourceBuilder{T}"/>.</returns>
+    public static IResourceBuilder<AzureServiceBusEmulatorResource> WithDataVolume(this IResourceBuilder<AzureServiceBusEmulatorResource> builder, string? name = null)
+        => builder.WithVolume(name ?? VolumeNameGenerator.Generate(builder, "data"), "/data", isReadOnly: false);
+
+    /// <summary>
+    /// Adds a bind mount for the configuration file of an Azure Service Bus emulator resource.
+    /// </summary>
+    /// <param name="builder">The builder for the <see cref="AzureServiceBusEmulatorResource"/>.</param>
+    /// <param name="path">Path to the file on the AppHost where the emulator configuration is located.</param>
+    /// <returns>A reference to the <see cref="IResourceBuilder{T}"/>.</returns>
+    public static IResourceBuilder<AzureServiceBusEmulatorResource> WithConfigJson(this IResourceBuilder<AzureServiceBusEmulatorResource> builder, string path)
+    => builder.WithBindMount(path, AzureServiceBusEmulatorResource.EmulatorConfigJsonPath, isReadOnly: false);
+
+    /// <summary>
+    /// Alters the JSON configuration document used by the emulator.
+    /// </summary>
+    /// <param name="builder">The builder for the <see cref="AzureServiceBusEmulatorResource"/>.</param>
+    /// <param name="configJson">A callback to update the JSON object representation of the configuration.</param>
+    /// <returns>A reference to the <see cref="IResourceBuilder{T}"/>.</returns>
+    public static IResourceBuilder<AzureServiceBusEmulatorResource> ConfigureJson(this IResourceBuilder<AzureServiceBusEmulatorResource> builder, Action<JsonNode> configJson)
+    {
+        ArgumentNullException.ThrowIfNull(builder);
+        ArgumentNullException.ThrowIfNull(configJson);
+
+        builder.WithAnnotation(new ConfigJsonAnnotation(configJson));
+
+        return builder;
+    }
+
+    /// <summary>
+    /// Configures the gateway port for the Azure Service Bus emulator.
+    /// </summary>
+    /// <param name="builder">Builder for the Azure Service Bus emulator container</param>
+    /// <param name="port">Host port to bind to the emulator gateway port.</param>
+    /// <returns>A reference to the <see cref="IResourceBuilder{T}"/>.</returns>
+    public static IResourceBuilder<AzureServiceBusEmulatorResource> WithGatewayPort(this IResourceBuilder<AzureServiceBusEmulatorResource> builder, int? port)
```

Comment:
Is "gateway port" the right term here?

Reply:
Consistency with the Event Hub and CosmosDB emulators.

Reply:
Did we use the term "gateway port" in CosmosDB, copy it to Event Hubs, and now it is going to propogate everywhere? CosmosDB has a "gateway" connection mode. 

In normal containers we call this "WithHostPort":

https://github.com/dotnet/aspire/blob/29a23f3c5eb089c30fc11fcc0b411244b2708a0a/src/Aspire.Hosting.PostgreSQL/PostgresBuilderExtensions.cs#L221
https://github.com/dotnet/aspire/blob/29a23f3c5eb089c30fc11fcc0b411244b2708a0a/src/Aspire.Hosting.Redis/RedisBuilderExtensions.cs#L327
https://github.com/dotnet/aspire/blob/29a23f3c5eb089c30fc11fcc0b411244b2708a0a/src/Aspire.Hosting.MongoDB/MongoDBBuilderExtensions.cs#L161